core-v1/src/main/scala/engine/StandardEngine.scala

```diff
@@ -126,22 +174,49 @@ object StandardEngine {
     ): PO <:< I => ExprResult[PO, I, Boolean, OP] = { implicit evPOisI =>
       val left = expr.leftExpr.visit(this)(implicitly)
       val right = expr.rightExpr.visit(this)(implicitly)
-      val output = left.state.output || right.state.output
+      val leftIsTrue = left.state.output
+      val rightIsTrue = right.state.output
+      val output = leftIsTrue || rightIsTrue
       // TODO: Do justification here
```

Comment:
Justification has moved to a `Justified` wrapper type, so it isn't part of the engine anymore. I should remove this TODO.